src/hypothesis/extra/datetime.py

```diff
@@ -70,6 +70,36 @@ def do_draw(self, data):
                 pass
 
 
+class TimedeltaStrategy(SearchStrategy):
+
+    def __init__(self, min_value=None, max_value=None):
+        MIN = dt.timedelta.min
+        MAX = dt.timedelta.max
+        # timedelta(0) is falsey, so we need to be explicit here
+        if min_value is None:
+            min_value = MIN
+        if max_value is None:
+            max_value = MAX
+        self.min_value = min_value
+        self.max_value = max_value
+
+    def do_draw(self, data):
+        result = dt.timedelta(
+            days=cu.centered_integer_range(
+                data, self.min_value.days, self.max_value.days, 0
+            ),
+            seconds=cu.integer_range(
+                data, self.min_value.seconds, self.max_value.seconds
+            ),
```

Comment:
This (and the microseconds version too) isn't actually the right logic. For example, if I have min_value=timedelta(days=0, seconds=40) then timdelta(days=1, seconds=0) is perfectly valid.

I wonder if you'd be better off doing the whole computation in microseconds - just draw an integer number of microseconds from the base and add that to self.min_value.